amd/src/initmathjax.js

```diff
@@ -1,22 +1,61 @@
-define(["jquery", "core/ajax"], function ($, Ajax) {
-  return {
-    init: function () {
-      $(document).ready(function () {
-        // Initialise MathJax typesetting
-        var nodes = Y.all(".latex-math");
-        Y.fire(M.core.event.FILTER_CONTENT_UPDATED, { nodes: nodes });
-
-        var waitForMathJax = setInterval(function () {
-          try {
-            if (MathJax) {
-              MathJax.Hub.Queue(["Typeset", MathJax.Hub]);
-              clearInterval(waitForMathJax);
+define(["jquery", "core/ajax", 'core_filters/events'], function ($, Ajax, FilterEvents) {
+
+    // Selectors and other strings defined here.
+    const SELECTORS = {
+        root: '.monassis',
+        mathEquation: '.latex-math',
+        classes: {
+            filterMathjax: 'filter_mathjaxloader_equation'
+        }
+    }
+
+
+    return {
+
+        init: function () {
+
+            // Parent element of the questions.
+            var root = document.querySelector(SELECTORS.root);
+
+            const updateMathJax = () => {
+                if (root.querySelectorAll(SELECTORS.mathEquation).length !== 0) {
+                    // Initialise MathJax typesetting.
+                    // Add additional filtermathjax class for fetch the list of nodes needed for filter.
+                    root.querySelectorAll(SELECTORS.mathEquation).forEach(
+                        (e) => e.classList.add(SELECTORS.classes.filterMathjax)
+                    );
+
+                    // Notifiy the siyavula filtered content are updated.
+                    FilterEvents.notifyFilterContentUpdated(root);
+
+                    // Ensure MathJax is ready before triggering typeset
+                    if (window.MathJax) {
+                        if (window.MathJax.Hub.queue.pending) {
+                            window.MathJax.Hub.Queue(["Typeset", window.MathJax.Hub, root]);
+                        }
+                    }
+                }
             }
-          } catch (e) {
-            // MathJax has not yet been loaded.
-          }
-        }, 100);
-      });
-    },
-  };
+
+            // Callaback to the observer.
+            function callback(mutationList, observer) {
+                // console.log('mutation', mutationList);
```

Comment:
@prasanna-lmsace I think you forgot to remove this.

Reply:
Removed the unused codes